### PR TITLE
Fix edit command not working with bikes

### PR DESCRIPTION
**Pull request by FongYuan**
For https://github.com/CS2103-AY1819S1-F10-2/main/issues/145

I've incidentally found that the `edit` command also does not work with `nric`, `rate`, `startTime` and `endTime`. They've all been fixed.

### CODE REVIEW

File: src/main/java/loanbook/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -184,7 +209,7 @@ public EditLoanDescriptor(EditLoanDescriptor toCopy) {
          * Returns true if at least one field is edited.
          */
         public boolean isAnyFieldEdited() {
-            return CollectionUtil.isAnyNonNull(name, phone, email, tags);
+            return CollectionUtil.isAnyNonNull(name, nric, phone, email, bike, rate, startTime, endTime, tags);
```
Comment (prokarius) [inline src/main/java/loanbook/logic/commands/EditCommand.java]: In fact, it is possible for endTime to be null, as this is the case when a loan is in progress.

**Comment by prokarius** [comment]
As per discussion, we are not allowing the editing of startTime and endTime. I would request that you undo the changes that you have made to allow the editing of these fields from the `edit` command.

**Comment by FongYuan** [comment]
> As per discussion, we are not allowing the editing of startTime and endTime. I would request that you undo the changes that you have made to allow the editing of these fields from the `edit` command.

Can do. Although the scenario I had in mind was that the user is intending to correct rather than correct the startTime and endTime. As with the other fields, say the user changes their mind after the loan has already been logged, and wants to loan like for example on Tue 2-8 pm rather than Mon 2-6 pm. (We haven't implemented bicycle rentals in advance yet, maybe never will for v1.4, but the example still stands for endTime.)

File: src/main/java/loanbook/logic/commands/EditCommand.java (hunk 1 above)
Comment (FongYuan) [reply src/main/java/loanbook/logic/commands/EditCommand.java]: This method is within the `EditLoanDescriptor` class. Which basically has a non-null field for any field that is going to be updated in the Loan. For example, I have an `EditLoanDescriptor` object whose members are all null except for the NRIC of "F1234567Z". What that means is that when the EditCommand executes, it'll only update the NRIC and no other field. There's no link to whether any data field such as `endTime` is required or optional in the Loan class over here.

**Pull request merged**